src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -116,18 +118,52 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
             var unmatchedPathInOriginalTemplate = pathInExpandedTemplate[originalTemplatePath.Length..].TrimStart('.');
 
             // Compare the path of the expanded template with the path of the JToken found in the original template.
-            // If they match, or if the first unmatched property in the original is NOT a resources array,
+            // If they match or if the first unmatched property in the original is NOT a resources array
+            // and it's looking in a resources array and the first unmatched property in the original is an array index
             // the line number from the JToken of the original template can be returned directly.
             if (originalTemplatePath.Length == pathInExpandedTemplate.Length
-                || !unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase))
+                || (!unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase) 
+                    && !unmatchedPathInOriginalTemplate.StartsWith("[", StringComparison.OrdinalIgnoreCase)
+                    && !originalTemplatePath.EndsWith("resources", StringComparison.OrdinalIgnoreCase)))
```

Comment:
See if you can make this a little more readable/understandable.  It might even be worth splitting this up into multiple `if`s.

Reply:
Is there a reason you compare the lengths of the paths here instead of comparing the strings themselves? `originalTemplatePath.Length == pathInExpandedTemplate.Length`

Reply:
If the paths between original and expanded match, doing a length comparison is faster than a string comparison.  Because the resource (or param, variable, etc.) properties are going to have to match in both paths, it shouldn't be possible to have these two strings be the same length but have different paths.

In looking into this though, this doesn't correctly account for different lengths in integer indexes in the resource array though (if the original resource of a copied one has to be looked up).  Before your change, it would still happen to work because it wouldn't be possible to cut off the string exactly before a *resources* property.

So short conclusion: this logic could use some updating.